app/services/katello/pulp3/content_view_version/metadata_map.rb

```diff
@@ -0,0 +1,119 @@
+module Katello
+  module Pulp3
+    module ContentViewVersion
+      class MetadataMap
+        class MetadataContentView < OpenStruct; end
+
+        class MetadataContentViewVersion < OpenStruct; end
+
+        class MetadataProduct < OpenStruct; end
+
+        class MetadataRepository < OpenStruct; end
+
+        class MetadataGpgKey < OpenStruct; end
+
+        class MetadataRepositoryContent < OpenStruct; end
+
+        class MetadataOrganization < OpenStruct; end
+
+        attr_reader :toc,
+                    :products,
+                    :repositories,
+                    :gpg_keys,
+                    :content_view,
+                    :content_view_version,
+                    :from_content_view_version
+
+        def initialize(metadata:)
+          @toc = metadata[:toc]
+          @products = parse_products(metadata[:products]) if metadata[:products]
+          @repositories = parse_repositories(metadata[:repositories]) if metadata[:repositories]
+          @content_view = parse_content_view(metadata[:content_view]) if metadata[:content_view]
+          @content_view_version = parse_content_view_version(metadata[:content_view_version]) if metadata[:content_view_version]
+          @from_content_view_version = parse_content_view_version(metadata[:from_content_view_version]) if metadata[:from_content_view_version]
+          @gpg_keys = parse_gpg_keys(metadata[:gpg_keys]) if metadata[:gpg_keys]
+        end
+
+        private
+
+        def parse_gpg_keys(gpg_keys)
+          gpg_keys.values.map do |g|
+            MetadataGpgKey.new(
+              name: g[:name],
+              content_type: g[:content_type],
+              content: g[:content]
+            )
+          end
+        end
+
```

Comment:
```ruby
        def parse_gpg_keys(gpg_keys)
          gpg_keys.values.map do |g|
            MetadataGpgKey.new(g)
          end
        end
```

Reply:
Kind of mixed on this approach. I don't find the need to elaborate every  parameter.

Reply:
> It's reasonable to me that just because we add some value in the exported metadata doesn't mean we want it exposed verbatim in the MetadataMap
I don't know if we need to be that picky. We trust what is on the disk is correct in any case. That being said if we make metadatamap also generate the json in a future pr, we can look to fixing that.